### ISSUE TITLE
Travis configuration and broken frameworks

### ISSUE THREAD

**Issue by nbrady-techempower** [TechEmpower/FrameworkBenchmarks#1838]
Currently, when editing a configuration or toolset file, travis will run through every framework and run a complete test. Broken frameworks eat up a ton of time and cause travis to fail. I suggest commenting out the frameworks in travis.yml that we know aren't working, so that if a complete test needs to be run it will save time and not cause contributers to panic, thinking they may have broken something. When the framework is fixed, it can be uncommented. 

Example:
````yml
    - "TESTDIR=Erlang/chicagoboss"
   # (Framework Not Working) - "TESTDIR=Erlang/cowboy"
    - "TESTDIR=Erlang/elli"
    - "TESTDIR=Go/beego"
    - "TESTDIR=Go/falcore"
    - "TESTDIR=Go/fasthttp"
````

If this is approved, I will go through and update.

**Comment by nbrady-techempower** [comment]
Note: Of course, it's preferable to fix either the framework if it's broken, the travis-ci configuration for that framework if it needs it, or remove the framework if it's no longer being supported in lieu of just commenting out frameworks and forgetting about them. The intention here is not to sweep things under the rug, but at the same time I think it's counter-intuitive (OHH That's what the ci in travis-ci stands for!) for travis to ever be failing on master, as master should probably always be ready (pull requests not withstanding) to run the next round of benchmarks.

That being said, it seems there may be fixes for the Scala issues in the PR's, the javascript frameworks fixed, ulib framework passing and the redis config patched. Also, I think this move could put us back on track for the workflow suggestion @LadyMozzarella made in #1437

**Comment by blee-techempower** [comment]
It will save tons of time, so I will disable all the broken frameworks from now.

**Comment by msmith-techempower** [comment]
I like this suggestion, but I would also like those comments in `.travis.yml` to have a link to an open issue describing the problem. Example:

```
    - "TESTDIR=Erlang/chicagoboss"
   # "TESTDIR=Erlang/cowboy" (Framework Not Working: https://github.com/TechEmpower/FrameworkBenchmarks/issues/1838)
    - "TESTDIR=Erlang/elli"
    - "TESTDIR=Go/beego"
    - "TESTDIR=Go/falcore"
    - "TESTDIR=Go/fasthttp"
```

**Comment by msmith-techempower** [comment]
One thing to note is that we (mostly @hamiltont) have spent a lot of time getting our travis scripts to run as quickly as possible by attempting to detect what has changed from the master branch to the pull request, and only test those frameworks that have changed.

Example: if a pull request only modified a single file in `frameworks/Java/gemini`, then every framework test in Travis immediately finishes as "successful" while Gemini will run its tests.

One of the problems is that we cannot yet (I have an idea for this) detect when a toolset item changes in conjunction with ONLY a framework.

Example: if a pull request only modified `toolsetup/setup/linux/languages/java7.sh`, then we have to test everything because there is not yet a simple way to determine which frameworks must be tested.

Just for scratch space, my idea leverages the `fw_depends` changes I made a while ago, but we could probably add more sophisticated checks to only run the frameworks that were modified (like in the first example) AND only run frameworks which specify dependencies via `fw_depends` when those dependencies are modified (like in the second example).

**Comment by nbrady-techempower** [comment]
Closed for #2074
